core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -6283,7 +6283,10 @@ protected function initRoots()
 		}
 
 		// Fetch all children of the root
-		$this->rootChildren = $this->Database->getChildRecords($this->root, $table, $this->Database->fieldExists('sorting', $table));
```

Comment:
This caused a DB error for tables without a `pid` column:

<img width="901" alt="" src="https://user-images.githubusercontent.com/1192057/145563992-b270b860-2cc9-43b3-b0c1-a49b544b968c.png">

@Toflar Since all tables with a `sorting` column also have a `pid` column in Contao, I have solved it this way. We could also check for a `pid` column in the if condition instead. WDYT?

Reply:
Same here, I see the problem but I don't think the solution is the right one :D It should be the same logic as everywhere else? Probably checking on the mode?

Reply:
Yes, we could check for `$this->treeView` as in line 6241. Or we could move line 6286 into the if condition above:

```php
if ($GLOBALS['TL_DCA'][$table]['list']['sorting']['showRootTrails'] ?? null)
{
	foreach ($this->root as $id)
	{
		$this->visibleRootTrails = array_unique(array_merge($this->visibleRootTrails, $this->Database->getParentRecords($id, $table, true)));
	}

	$this->rootChildren = $this->Database->getChildRecords($this->root, $table, $this->Database->fieldExists('sorting', $table));
}
```

Or is `$this->rootChildren` independent from the `showRootTrails` setting?

Reply:
I think it's independent. It just contains the children of `root` which is a permission thing. `showRootTrails` is not related to permissions directly. Imho it's dependent on the tree mode. If we're not in the tree mode, we don't have any `root` children, right?

Reply:
Fixed in 0a30a0026a9881aa5d16e2625ad2e846542ecfba.